### PR TITLE
add logout

### PR DESCRIPTION
**Pull request by Abecker37**
## Description

add logout feature

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [ ] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [x] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [x] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

File: packages/frontend/web/src/components/Navbar.tsx [hunk 1]
```diff
@@ -160,6 +160,33 @@ export default function Navbar() {
...
+                            const response = await fetch(
+                              `${import.meta.env.VITE_API_URL}/auth/logout`,
+                              { method: 'POST', credentials: 'include' },
+                            );
+
+                            if (response.ok) {
+                              setIsLoggedIn(false);
+                              <Navigate to='/login' />;
+                            } else {
+                              console.error('Échec de la déconnexion');
```
Comment (JorisPineau) [inline packages/frontend/web/src/components/Navbar.tsx:176]: "Échec de la déconnexion"

File: packages/frontend/web/src/components/Navbar.tsx [hunk 2]
```diff
@@ -160,6 +160,33 @@ export default function Navbar() {
...
+                            }
+                          } catch (error) {
+                            console.error(
+                              'Erreur lors de la déconnexion',
```
Comment (JorisPineau) [inline packages/frontend/web/src/components/Navbar.tsx:180]: 'Erreur lors de la déconnexion'

File: packages/backend/api/src/auth.ts [hunk 3]
```diff
@@ -171,5 +171,14 @@ authRouter.post(
...
   },
 );
+authRouter.post('/logout', (req, res) => {
+  res.cookie('token', '', {
+    httpOnly: true,
+    secure: IS_PRODUCTION,
+    sameSite: 'lax',
+    expires: new Date(0),
+  });
+  res.status(200).send('Déconnecté avec succès');
```
Comment (JorisPineau) [inline packages/backend/api/src/auth.ts:181]: 'Déconnecté avec succès' en anglais

**Review by JorisPineau — CHANGES_REQUESTED** [review]

**Review by herve-perel — APPROVED** [review]

**Review by JorisPineau — APPROVED** [review]

File: packages/backend/api/src/auth.ts [hunk 4]
```diff
@@ -171,5 +171,14 @@ authRouter.post(
...
   },
 );
+authRouter.post('/logout', (req, res) => {
+  res.cookie('token', '', {
+    httpOnly: true,
+    secure: IS_PRODUCTION,
+    sameSite: 'lax',
+    expires: new Date(0),
+  });
+  res.status(200).send('Logout Sucessful');
```
Comment (YDrogen) [inline packages/backend/api/src/auth.ts:181]: C'est pas du JSON ça

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/backend/api/src/auth.ts (hunk 4 above)
Comment (Abecker37) [reply packages/backend/api/src/auth.ts:181]: Tu peux être plus clair ? il faut que j'enlève le res.status ?  c'est mon Token qui est pas bon ?

**Review by JorisPineau — APPROVED** [review]

**Review by bugBuster42 — APPROVED** [review]

File: packages/backend/api/src/auth.ts (hunk 4 above)
Comment (YDrogen) [reply packages/backend/api/src/auth.ts]: Il faut quand même envoyer une réponse, sinon le client va attendre indéfiniment
Mais au format JSON, sinon le client va planter

**Review by JorisPineau — APPROVED** [review]

**Review by bugBuster42 — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**